src/test/java/jp/ikedam/jenkins/plugins/extensible_choice_parameter/ExtensibleChoiceParameterDefinitionJenkinsTest.java

```diff
@@ -1351,4 +1358,86 @@ public void testComboboxFilter() throws Exception
         );
         in.blur();
     }
+
+    private XmlPage getXmlPage(WebClient wc, String path) throws Exception
+    {
+        // WebClient#getPageXml expects content-type "application/xml"
+        // and doesn't accept "text/xml".
+        Page p = wc.goTo(path, null);
+        if (p instanceof XmlPage) {
+            return (XmlPage)p;
+        }
+        throw new AssertionError(
+            "Expected XML but instead the content type was "
+                + p.getWebResponse().getContentType()
+        );
+    }
+
+    @Test
+    public void testRestApiWithPermission() throws Exception
+    {
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        j.jenkins.setAuthorizationStrategy(new MockAuthorizationStrategy()
+                .grant(Item.READ, Item.BUILD, Jenkins.READ).everywhere().to("user")
+        );
+
+        FreeStyleProject p = j.createFreeStyleProject();
+        ExtensibleChoiceParameterDefinition def = new ExtensibleChoiceParameterDefinition(
+            "test",
+            new MockChoiceListProvider(
+                Arrays.asList("foo", "bar", "baz"),
+                null
+            ),
+            true,
+            "description"
+        );
+        p.addProperty(new ParametersDefinitionProperty(def));
+
+        WebClient wc = j.createWebClient();
+        wc.login("user");

```

Comment:
Actually, REST api is expected to use with BASIC authentication (https://www.jenkins.io/doc/book/using/remote-access-api/#RemoteaccessAPI-RemoteAPIandsecurity).

I haven't tested that with actual Jenkins instance, using BASIC authentication
(I have trouble in local Jenkins for SSL validations...).

Reply:
I couldn't see the permission behavior with `mvn hpi:run`. It always returns choices even to users without `Item.BUILD` permssions. REST API might always work with SYSTEM permission, or there might have been a security issue with older Jenkins.

Reply:
I verified that this works perfect with the latest LTS (2.249.2) and also with `mvn hpi:run`.
(It looks that I misconfigured permissions last time).